tools: fix warning from scan-build static code analyzer

Fix the following warning from the scan-build static code analyzer:

scan-build ./configure
scan-build --status-bugs make
. . .
  CC       tools/tpm2_import.o
tools/tpm2_import.c:308:33: warning: Result of 'malloc' is converted to a pointer of type 'uint8_t', which is incompatible with sizeof operand type 'TPM2B_PUBLIC'
    uint8_t *marshalled_bytes = malloc(sizeof(ctx.import_key_public));
    ~~~~~~~~~                   ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
  CCLD     tools/tpm2_import
 
This is in addition to scan-build warnings fixed with the fix to issue #983.

--- a/tools/tpm2_import.c
+++ b/tools/tpm2_import.c
@@ -305,21 +305,22 @@ static bool create_import_key_public_data_and_name(void) {
             ctx.import_key_public_unique_data, TPM2_SHA256_DIGEST_SIZE);
 
     size_t public_area_marshalled_offset = 0;
-    uint8_t *marshalled_bytes = malloc(sizeof(ctx.import_key_public));
+    TPM2B_PUBLIC *marshalled_bytes = malloc(sizeof(ctx.import_key_public));
     if (!marshalled_bytes) {
         LOG_ERR("oom");
         return false;
     }
 
-    Tss2_MU_TPM2B_PUBLIC_Marshal(&ctx.import_key_public, marshalled_bytes,
-            sizeof(ctx.import_key_public), &public_area_marshalled_offset);
+    Tss2_MU_TPM2B_PUBLIC_Marshal(&ctx.import_key_public,
+        (uint8_t *)marshalled_bytes, sizeof(ctx.import_key_public),
+        &public_area_marshalled_offset);
 
     ctx.import_key_public_name.size = TPM2_SHA256_DIGEST_SIZE;
     size_t name_digest_alg_offset = 0;
     Tss2_MU_UINT16_Marshal(TPM2_ALG_SHA256, ctx.import_key_public_name.name,
             sizeof(TPM2_ALG_ID), &name_digest_alg_offset);
     ctx.import_key_public_name.size += name_digest_alg_offset;
-    SHA256(marshalled_bytes + sizeof(uint16_t),
+    SHA256((uint8_t *)marshalled_bytes + sizeof(uint16_t),
             public_area_marshalled_offset - sizeof(uint16_t),
             ctx.import_key_public_name.name + sizeof(TPM2_ALG_ID));
     free(marshalled_bytes);

